docs/coding-guidelines.md

```diff
@@ -0,0 +1,402 @@
+# Coding guidelines
+
+Let's face it. No matter what coding guidelines we choose, we're not going to make everyone happy.
+In fact, some people out there might be downright angry at the choices we make.
+But the fact of the matter is that there is no "one true bracing style," despite [attempts to name a bracing style as such](http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS). 
+
+While we would like to embrace everyone's individual style, working together on the same codebase would be utter chaos if we don't enforce some consistency. When it comes to coding guidelines, consistency can be even more important than being "right."
+
+## Copyright header and license notice
+
+All source code files (mostly `src/**/*.cs` and `test/**/*.cs`) require this exact header (please do not make any changes to it):
+
+```cs
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+```
+
+It is ok to skip it on generated files, such as `*.designer.cs`.
```

Comment:
It is not mandatory to add the license header to Auto-generated files , such as `*.designer.cs`